src/wallet/wallet.cpp

```diff
@@ -2460,44 +2459,20 @@ TransactionError CWallet::FillPSBT(PartiallySignedTransaction& psbtx, bool& comp
                 input.non_witness_utxo = wtx.tx;
             }
         }
+    }
 
-        // Get the Sighash type
-        if (sign && input.sighash_type > 0 && input.sighash_type != sighash_type) {
-            return TransactionError::SIGHASH_MISMATCH;
-        }
-
-        // Get the scriptPubKey to know which SigningProvider to use
-        CScript script;
-        if (!input.witness_utxo.IsNull()) {
-            script = input.witness_utxo.scriptPubKey;
-        } else if (input.non_witness_utxo) {
-            if (txin.prevout.n >= input.non_witness_utxo->vout.size()) {
-                return TransactionError::MISSING_INPUTS;
-            }
-            script = input.non_witness_utxo->vout[txin.prevout.n].scriptPubKey;
-        } else {
-            // There's no UTXO so we can just skip this now
-            complete = false;
-            continue;
-        }
-        SignatureData sigdata;
-        input.FillSignatureData(sigdata);
-        std::unique_ptr<SigningProvider> provider = GetSigningProvider(script, sigdata);
-        if (!provider) {
-            complete = false;
-            continue;
+    // Fill in information from ScriptPubKeyMans
+    for (const auto& spk_man_pair : m_spk_managers) {
+        TransactionError res = spk_man_pair.second->FillPSBT(psbtx, sighash_type, sign, bip32derivs);
```

Comment:
In a90a98c3aa90ed4dc4f995fac0859e58f177a79a under `// Fill in information from ScriptPubKeyMans`, add `CanProvide()`?

Reply:
I've updated it to use `GetScriptPubKeyMan` which calls `CanProvide`

This change is slightly more complex but it does reduce the number of repeated signings.